master/getting-started/openstack/upgrade/index.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Upgrading Calico for OpenStack
+canonical_url: 'https://docs.projectcalico.org/master/getting-started/openstack/upgrade/'
+---
+## Upgrade procedure (OpenStack)
+These documents detail the procedure for upgrading {{site.prodname}} in an OpenStack system. 
+If a release does not upgrade a particular component, you should skip the steps for that component.
+
+> **Important**: You will
+> be unable to issue API requests to your OpenStack system during the
+> procedure. Please plan your upgrade window accordingly, and see the
+> [Service Impact](#service-impact) section for more details.
+{: .alert .alert-danger}
+
+## Service impact
+
+During the upgrade, **all VMs will continue to function normally**:
+there should be no impact on the data plane. However, control plane
+traffic may fail at different points throughout the upgrade.
+
+Generally, users should be prevented from creating or updating virtual
+machines during this procedure, as these actions will fail. VM deletion
+*may* succeed, but will likely be delayed until the end of the upgrade.
+
+For this reason, we highly recommend planning a maintenance window for
+the upgrade. During this window, you should disable all user API access
+to your OpenStack deployment.
+
+
+## Before you begin
+
+- You must first [upgrade](/v2.6/getting-started/openstack/upgrade) 
+  to {{site.prodname}} [v2.6.x](https://github.com/projectcalico/calico/releases) 
+  before you can upgrade to {{site.prodname}} 
+  {{site.data.versions[page.version].first.title}}. 
+
+## Upgrade steps
+
+If you use calicoctl to add Calico policies in addition to those that are derived automatically (by our Neutron driver) from OpenStack security groups - or more generally to configure any Calico data other than the data that is generated by the Calico Neutron driver - you will need to follow the steps marked OPTIONAL below. Otherwise you can skip those OPTIONAL steps.
+
+1. Set up an etcdv3 cluster, if you don't already have etcdv3 capability, either by [upgrading the etcd software of your existing etcdv2 cluster](https://coreos.com/etcd/docs/latest/upgrades/upgrade_3_0.html), or by [installing an etcdv3 cluster on new servers](https://coreos.com/etcd/docs/latest/op-guide/clustering.html).
+  
+   > **Tip**: etcd v3.x still supports the etcd v2 API, as used by {{site.prodname}} v2.6.x.
+   > However, {{site.prodname}} v3.0.0+ does not support the etcd v2 API.  The upgrade steps below
+   > move the data from the etcd v2 API to the etcd v3 API.  This requires an etcd v3.x server.
+   {: .alert .alert-success}
+  
+1. **OPTIONAL Set the version flag.**  On the control node:
+   ```
+   etcdctl set /calico/v1/config/CalicoVersion v2.6.5-7
```

Comment:
I am noticing here that you explicitly have the Calico version at v2.6.5-7, yet above, you had `2.6` or later... Actually, we are on 2.6.8 now it looks like.

Reply:
This is actually a nasty hack.  calico-upgrade will not operate unless it sees this key set in etcd.  Openstack doesn't actually have or need this key at all.  

The only reason we set this is to make the calico-upgrade tool work.  In that sense, its irrelevant what the value is, so long as its a value that will allow the calico-upgrade tool to work.

Once we move up to the etcdv3 and delete the etcdv2 data, this key will be gone.